### PR TITLE
Fixes #778 Import optimization

### LINKED ISSUE #778: Optimize import workflow
### Description

- Move `xASL_..._Bids2Legacy` to `xASL_module_Import`
- Make `xASL_imp_DCM2NII_Initialize` conditional
- Introduce subfunctions in `xASL_..._Bids2Legacy`

### Release notes

Optimize import workflow for code simplification and robustness.

### PR DESCRIPTION
**Pull request by MichaelStritt**
### Linked issue

#778

### Comment

@jan-petr: In some of the commits I added extra detailed comments on what I did. I think this overall improves the import a lot. I basically did the changes that I described in the WhatsApp group (related to the flow charts) and also fixed some bugs related to #769.

### Simplified description

- Move `xASL_..._Bids2Legacy` to `xASL_module_Import`
- Make `xASL_imp_DCM2NII_Initialize` conditional / rename it
- Introduce subfunctions in `xASL_..._Bids2Legacy`
- Fix bugs related to #769

### CODE REVIEW

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy.m [hunk 1]
```diff
@@ -10,7 +10,7 @@
 %   dataPar    - dataPar values to be filled in a basic dataPar created with the conversion to legacy 
 %                (OPTIONAL, DEFAULT = basic dataPar settings)
 %   
-% OUTPUT: n/a
+% OUTPUT: dataPar - dataPar struct (STRUCT)
```
Comment (jan-petr) [inline Modules/SubModule_Import/xASL_bids_BIDS2Legacy.m:13]: Line 4 - FORMAT - x missing

File: Modules/SubModule_Import/xASL_bids_AddGeneratedByField.m [hunk 2]
```diff
@@ -0,0 +1,42 @@
+function xASL_bids_AddGeneratedByField(x, pathJSONin, pathJSONout)
+%xASL_bids_AddGeneratedByField Add the generated by field to the struct.
+%
+% FORMAT:     xASL_bids_AddGeneratedByField(x, pathJSONin, pathJSONout)
+% 
+% INPUT:      x           - ExploreASL x struct (STRUCT, REQUIRED)
+%             pathJSONin  - Path to input JSON file (STRING, REQUIRED)
+%             pathJSONout - Path to output JSON file (STRING, REQUIRED)
```
Comment (jan-petr) [inline Modules/SubModule_Import/xASL_bids_AddGeneratedByField.m:8]: This last variable is not required. If nargin =2, then overwrite the input - see line 22 and modify header accordingly.

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    Reference{iScan+1, 3} = 1; % default to 1st run
+                end
+            end
+            Reference(2:end,:) = sortrows(Reference(2:end,:), [2, 3]); % first sort for ScanType then run
+
+            RunsAre = cellfun(@(y) y, Reference(2:end, 3));
+            RunsUnique = unique(RunsAre);
+
+
+            %% 4.1. Parse scantype
```
Comment (jan-petr) [inline Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m:47]: remove 4.1

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m [hunk 4]
```diff
@@ -0,0 +1,59 @@
+function xASL_bids_BIDS2Legacy_ParseModality(BIDS, bidsPar, SubjectVisit, iSubjSess, ModalitiesUnique, nModalities, bOverwrite, pathLegacy_SubjectVisit)
+%xASL_bids_BIDS2Legacy_ParseModality Parse modality for BIDS to Legacy conversion.
+%
+% FORMAT:     xASL_bids_BIDS2Legacy_ParseModality(BIDS, bidsPar, SubjectVisit, iSubjSess, ModalitiesUnique, nModalities, bOverwrite)
```
Comment (jan-petr) [inline Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m:4]: sync format and function header.

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_CompilePathsForCopying.m [hunk 5]
```diff
@@ -0,0 +1,75 @@
+function [bidsPar, TypeIs, pathOrig, pathDest] = xASL_bids_BIDS2Legacy_CompilePathsForCopying(bidsPar, TypeIs, ModalityIs, RunIs, iSubjSess, BIDS, TypeRunIndex, ModalityFields, pathLegacy_SubjectVisit)
+%xASL_bids_BIDS2Legacy_CompilePathsForCopying Compile paths for BIDS to Legacy copying.
+%
+% FORMAT:     [bidsPar, TypeIs, pathOrig, pathDest] = xASL_bids_BIDS2Legacy_CompilePathsForCopying(bidsPar, TypeIs)
```
Comment (jan-petr) [inline Modules/SubModule_Import/xASL_bids_BIDS2Legacy_CompilePathsForCopying.m:4]: sync format with function header

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseScanType.m [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            for iRun=1:length(RunsUnique) % iterate runs in this Subject/Visit/Modality
+                RunIs = RunsUnique(iRun);
+                RunIndices = RunsAre==RunsUnique(iRun);
+                TypeRunIndex = find(RunIndices & TypeIndices);
+                if length(TypeRunIndex)>1
+                    warning(['Multiple NIfTIs found for ' SubjectVisit '_run-' xASL_num2str(RunIs) '_' TypeIs ', using first only']);
+                    TypeRunIndex = TypeRunIndex(1);
+                end
+
+                %% 4.2. Compile paths for copying
```
Comment (jan-petr) [inline Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseScanType.m:44]: remove 4.2, 4.3, 4.4

**Review by jan-petr — CHANGES_REQUESTED** [review]
Minor admin stuff. Couldn't discover large issues.

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy.m (hunk 1 above)
Comment (MichaelStritt) [reply Modules/SubModule_Import/xASL_bids_BIDS2Legacy.m:13]: Oh sorry, I was updating the IO parameters multiple times and forgot to update the format some of the times. Should be fixed in 77a1e39 👍

File: Modules/SubModule_Import/xASL_bids_AddGeneratedByField.m (hunk 2 above)
Comment (MichaelStritt) [reply Modules/SubModule_Import/xASL_bids_AddGeneratedByField.m]: Easy fix 77a1e39 👍

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m (hunk 3 above)
Comment (MichaelStritt) [reply Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m]: Removed 77a1e39

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m (hunk 4 above)
Comment (MichaelStritt) [reply Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseModality.m]: Synced 77a1e39

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_CompilePathsForCopying.m (hunk 5 above)
Comment (MichaelStritt) [reply Modules/SubModule_Import/xASL_bids_BIDS2Legacy_CompilePathsForCopying.m]: Synced 77a1e39

File: Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseScanType.m (hunk 6 above)
Comment (MichaelStritt) [reply Modules/SubModule_Import/xASL_bids_BIDS2Legacy_ParseScanType.m]: Removed 77a1e39

**Review by jan-petr — APPROVED** [review]
OK

**Pull request merged**